src/client/components/fields/DateField.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { FC, useState } from 'react'
+import { useFormContext } from 'react-hook-form'
+import DatePicker from 'react-datepicker'
+import {
+  FormControl,
+  FormLabel,
+  FormHelperText,
+  FormErrorMessage,
+  Input,
+} from '@chakra-ui/react'
+
+import 'react-datepicker/dist/react-datepicker.css'
+import '../../styles/date-picker.css'
+
+export interface DateProps {
+  order: number
+  id: string
+  description: string
+  help?: string
+  newDate: Date
+}
+
+export const DateField: FC<DateProps> = ({ order, id, description, help }) => {
+  const { register, errors } = useFormContext()
+  const [date, setDate] = useState(new Date())
+  const error = errors[id]
+
+  const changeHandler = (newDate: Date) => {
+    if (newDate) setDate(newDate)
+  }
+
+  return (
+    <FormControl isInvalid={error} id={id}>
+      <FormLabel htmlFor={id}>{`${order + 1}. ${description}`}</FormLabel>
+      {help && <FormHelperText mb={4}>{help}</FormHelperText>}
+      <Input
+        type="hidden"
+        name={id}
+        value={date && date.toISOString().split('T')[0]}
```

Comment:
Since variables will be passed into the final formula through string substitution, might be easier if we represent date as milliseconds since epoch. Avoid having to parse the date format again when evaluating for the result.

Reply:
Good point, updated in https://github.com/opengovsg/checkfirst/pull/12/commits/1d65016a3ee19d586db3bfe0d06c5f3426cecb6e